node/src/database/db_migrations.rs

```diff
@@ -74,353 +158,684 @@ impl DbMigratorReal {
         let connection = Connection::open_in_memory().unwrap();
         connection
             .execute(
                 "CREATE TABLE test (
-                name TEXT,
-                value TEXT
-            )",
+            name TEXT,
+            value TEXT
+        )",
                 NO_PARAMS,
             )
             .unwrap();
-        let connection_wrapper = ConnectionWrapperReal::new(connection);
-        let outdated_schema = "0.0.10";
-        let statement = "INSERT INTO test (name, value) VALUES (\"booga\", \"gibberish\")";
-        let list = &[&FakeMigrationRecord::default()
-            .version_result("0.0.10")
-            .sql_statement(statement) as &(dyn MigrateDatabase + 'static)];
-        let subject = DbMigratorReal::default();
-
-        let result = subject.make_updates(outdated_schema, list, Box::new(connection_wrapper));
+        connection
+            .execute(
+                "INSERT INTO test (name, value) VALUES ('schema_version', '3')",
+                NO_PARAMS,
+            )
+            .unwrap();
+        let mut connection_wrapper = ConnectionWrapperReal::new(connection);
+        let config = DBMigratorConfiguration {
+            db_configuration_table: "test".to_string(),
+        };
+        let subject = DbMigratorReal::new();
 
-        assert!(result.is_ok());
-        TestLogHandler::new().exists_log_containing(
-            "INFO: DbMigrator: Database successfully updated from version 0.0.10 to 0.0.11",
+        let result = subject.make_updates(
+            2,
+            5,
+            Box::new(DBMigrationUtilitiesReal::new(&mut connection_wrapper, config).unwrap()),
+            list_of_updates,
         );
+
+        assert_eq!(result, Ok(()));
+        let first_record_old_version_param = first_record_old_version_p_arc.lock().unwrap();
+        assert_eq!(first_record_old_version_param.len(), 1);
+        let second_record_old_version_param = second_record_old_version_p_arc.lock().unwrap();
+        assert_eq!(second_record_old_version_param.len(), 1);
+        let third_record_old_version_param = third_record_old_version_p_arc.lock().unwrap();
+        assert_eq!(third_record_old_version_param.len(), 2);
+        let fourth_record_old_version_param = fourth_record_old_version_p_arc.lock().unwrap();
+        assert_eq!(fourth_record_old_version_param.len(), 2);
+        let fifth_record_old_version_param = fifth_record_old_version_p_arc.lock().unwrap();
+        assert_eq!(fifth_record_old_version_param.len(), 3);
+        let first_record_migration_params = first_record_migration_p_arc.lock().unwrap();
+        assert_eq!(*first_record_migration_params, vec![]);
+        let second_record_migration_params = second_record_migration_p_arc.lock().unwrap();
+        assert_eq!(*second_record_migration_params, vec![]);
+        let third_record_migration_params = third_record_migration_p_arc.lock().unwrap();
+        assert_eq!(*third_record_migration_params, vec![()]);
```

Comment:
This is confusing, given that the database holds schema version 3. If this is just an artifact of testing, then maybe change the schema version back to 2 to make it clear?